### DISCUSSION TITLE
Input Shaping Setup Guide

### DISCUSSION THREAD

**Discussion by BEprints** [mriscoc/Ender3V2S1#580] [Show and tell]
I have put this together to hopefully help some people set up Input shaping included is a brief guide with the steps to take and links to helpful sites, Klippers RingingTower STL, 3 test Gcodes files for 0.4mm Nozzle I also complied firmware for 4.2.2. and 4.2.7 with Linear Advance and Input Shaping.

Any input or suggestions welcome as I am not the best at this kind of thing.

Feel free to do what you wish with this I just wanted to hopefully help some people as there isn't much documentation out there for Input Shaping just yet.

[Input Shaping Setup Guide For Mriscoc firmware](https://drive.google.com/drive/folders/1ictVUtGV0Du4BCcaipUk0jAiqZLNinuh?usp=sharing)

![bENCHY](https://user-images.githubusercontent.com/115055399/209632677-77437684-a7b7-48f6-ae57-9403dd7c75ef.jpg)

There is now a grate video out by Teaching Tech that does a good job of explaining things so please watch
https://www.youtube.com/watch?v=7VLuz3hReYw

**Comment by RobbieRw** [comment]
Do you have a gcode file for the tower please?

**Reply by BEprints** [reply]
There is some in the files if they don't work for your set up you can slice your own using Prusa slicer

**Comment by BEprints** [comment]
I have made some changes as there is now a plugin for Cura I have been able to get working but have not tested with prints

**Comment by dfinberg** [comment]
I've played with it just a bit, enough to do a first pass at calibration.  Trying some test prints now. Today is evidently fiddle with the printer day.

The provided gcodes have the temp reset at layer 1, which is something to be aware of if you need to bump the temp a bit.  The tough PLA I was using is printable at 210 with normal speeds, but at high speeds it needed to be up at 215.  I tried fiddling with cura and somehow completely borked my settings but was able to reset things eventually.

**Reply by BEprints** [reply]
I'll have to check out the gcode again I did try make it pretty standard as print temp can be tuned via the screen but the start gcode the send goes up to 150 I think with bed probing and then parks on the side and finishes heating to the print temp.

I am always happy for some feedback on how to improve this guide I am also hoping to do some testing with the cura plugin this weekend.

I have been learning Adobe Acrobat so will try make a pdf with screenshots and photos to make it a bit easier to understand.

**Reply by dfinberg** [reply]
If you do use cura, add the post-processing script of gcode documentation to your setup. That makes it easier for everyone else to reconstruct what you did with just the gcode.

The problem on heating seems to be extra M104's in the file, for example in file 3, there is
`M117 Getting the extruder up to temp!23
M104 S230.0 ; Set Extruder temperature
M109 S230.0 ; Wait for Extruder temperature
...
prints layer 0
...
G0 F12000 X85.084 Y60.76
M104 S230
G1 F1500 X86.564 Y62.239 E141.82289
G1 F2700 E136.82289
;MESH:NONMESH
G0 F300 X86.564 Y62.239 Z0.4
G0 F12000 X62.8 Y140.63
;TIME_ELAPSED:219.645094
;LAYER:1
;MESH:ringing_tower.stl
`

because of that if you just correct the temp as it starts doing actual printing it will reset when doing the actual test part.

Anyways, thanks for doing this, IS was on my list of things to try out with Klipper but I hadn't gotten around to getting it all installed yet. I guess I need to upgrade fans and ducts but that's more of a pain for a neo. I'm pretty clean at 100mm/s on the Y plane, but still seeing some ringing on the X. I guess I will try and redo parameters tonight and see what I learn.

**Reply by BEprints** [reply]
; Ender 3 Custom Start G-code
M104 S150; start warming extruder to 150
M140 S{material_bed_temperature_layer_0} ; Set Heat Bed temperature
M190 S{material_bed_temperature_layer_0} ; Wait for Heat Bed temperature
G28 ; Home all axes
G29 A ; Activate bed mesh -level (CR-Touch)
G29 L0 ; Load mesh from slot 0
G29 J3 ; Probe for bed tilt
G1 X245 Y218.5 Z1 F5000.0
G92 E0 ; Reset extruder
M117 Getting the extruder up to temp!
M104 S{material_print_temperature_layer_0} ; Set Extruder temperature
M109 S{material_print_temperature_layer_0} ; Wait for Extruder temperature
G1 Z1.0 F3000 ; move z up little to prevent scratching of surface
G1 X225 Y200 Z0.3 F5000.0 ; move to start-line position
G1 X225 Y20 Z0.3 F1500.0 E15 ; draw 1st line
G1 X222.2 Y20 Z0.3 F5000.0 ; move to side a little
G1 X222.2 Y200 Z0.3 F1500.0 E30 ; draw 2nd line
G92 E0 ; reset extruder
G1 Z1.0 F3000 ; move z up little to prevent scratching of surface
; End of custom start GCode


This is the start Gcode used it is designed to preheat the nozzle at the same time as heating the bed and bed probing
It first got to 150 then after probing it will park the head and finished heating to the temperature set in Cura

The speed test is easily replicated in Cura just use the ringing tower stl and the Auto towers plugin for the speed scrip but if you want you can edit the gcode yourself with notepad replace the start gcode with your own.

I set the temp to 230c because at 100mm/s there can be flow issues at the colder temps so if the extruder starts o click at high speed then bump the temp.

**Reply by BEprints** [reply]
https://www.youtube.com/watch?v=7VLuz3hReYw

**Comment by RobbieRw** [comment]
Awesome news!! Lets hope we get input shaping and the speed of the ender 3
V2 to the max :D

**Reply by BEprints** [reply]
The fastest you can print on a stock ender 3 is not very fast as you run into issues with layer cooling but after fixing that issue you can print uptown 100mm/s then start hitting the limitations of the hot end if you upgrade the hotend you can print faster I have a Spider hotend and have printed at 150mm/s and had no issues, I would recommend the Spider V3 over any other Spider hotend

**Comment by BEprints** [comment]
https://www.youtube.com/watch?v=7VLuz3hReYw

**Comment by mechano** [comment]
> I've played with it just a bit, enough to do a first pass at calibration. Trying some test prints now. Today is evidently fiddle with the printer day.
> 
> The provided gcodes have the temp reset at layer 1, which is something to be aware of if you need to bump the temp a bit. The tough PLA I was using is printable at 210 with normal speeds, but at high speeds it needed to be up at 215. I tried fiddling with cura and somehow completely borked my settings but was able to reset things eventually.

With CHT nozzle you don't need to rise temperature of PLA.
